fees-margins-estimation/wallet-helper.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"strings"
+)
+
+type WalletConfig struct {
+	URL        string `json:"URL"`
+	Passphrase string `json:"passphrase"`
+	Name       string `json:"Name"`
+}
+
+type Token struct {
+	Token string `json:"token"`
+}
+
+type Keys struct {
+	Keys []struct {
+		Pub     string `json:"pub"`
+		Algo    string `json:"algo"`
+		Tainted bool   `json:"tainted"`
+		Meta    []struct {
+			Key   string `json:"key"`
+			Value string `json:"value"`
+		} `json:"meta"`
+	} `json:"keys"`
+}
+
+var chars = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		v, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		b[i] = chars[v.Int64()]
+	}
+	return string(b)
+}
+
+func CheckUrl(url string) bool {
+	return url != "" && (strings.HasPrefix(url, "https://") || strings.HasPrefix(url, "http://"))
+}
+
+func CheckWalletUrl(url string) string {
+	suffix := []string{"/api/v1/", "/api/v1", "/"}
+	for _, s := range suffix {
+		if strings.HasSuffix(url, s) {
+			fmt.Println("There's no need to add ", s, " to WALLETSERVER_URL. Removing it.")
+			return url[0 : len(url)-len(s)]
+		}
+	}
+	return url
+}
+
+func CreateWallet(config WalletConfig) ([]byte, error) {
+	// __create_wallet:
+	// Create a new wallet:
+	jsonStr := []byte("{\"wallet\":\"" + config.Name + "\",\"passphrase\":\"" + config.Passphrase + "\"}")
```

Comment:
Consider using `json.Marshal()` here. (Also, imagine someone using `wallet", "inj":"ection` as their wallet name :smile: )

Something like:

```go
import "code.vegaprotocol.io/go-wallet/wallet"

// ...

req := &wallet.CreateLoginWalletRequest{Wallet: config.Name, Passphrase: config.Passphrase}
payload, err := json.Marshal(req)
if err != nil {
	// handle
}
```

See [go-wallet/wallet/service.go](https://github.com/vegaprotocol/go-wallet/blob/fadb3ad89367bbaa7f7dc99437f9c360aa70857d/wallet/service.go#L31-L35) for details.

Reply:
I've used the types defined in go-wallet. I didn't find them before :)